code-challenges.test.js

```diff
@@ -23,26 +23,63 @@ var colors1 = ["purple", "blue", "green", "yellow", "pink"]
 var colors2 = ["chartreuse", "indigo", "periwinkle", "ochre", "aquamarine", "saffron"]
 // Expected output example (can be a different order): ["saffron", "aquamarine", "periwinkle", "indigo", "ochre"]
 
-
+describe("shuffleArray", () => {
+    var colors1 = ["purple", "blue", "green", "yellow", "pink"]
+    var colors2 = ["chartreuse", "indigo", "periwinkle", "ochre", "aquamarine", "saffron"]
+    test("removes the first item from the array and shuffles the remaining content.", () => {
+      expect(shuffleArray(colors1)).toEqual(expect.arrayContaining(["green", "blue", "yellow", "pink"]))
+      expect(shuffleArray(colors2)).toEqual(expect.arrayContaining(["saffron", "aquamarine", "periwinkle", "indigo", "ochre"])
+    )})
+  })
+//At first, I tried using my function logic inside of the test expect.toEqual() method but kept getting an error.
+//I also tried using colors1.includes() inside of that method but got an error because that only returns a boolean value.
+//After some hardcore googling I found the jestjs.io website with 50+ different expect methods!
+//The method that ended up working was expect.arrayContaining()
 
 // b) Create the function that makes the test pass.
 
-
+const shuffleArray = (array) => {
+    let shuffled = array.slice(1, array.length)
+    .map((value) => ({ value, sort: Math.random() }))
+    .sort((a, b) => a.sort - b.sort)
+    .map(({ value }) => value)
+    return shuffled
+}
+shuffleArray(colors1)
+shuffleArray(colors2)
+//For the function I used a slice() method on the array to essentially remove the first value. 
+//I followed that with an array shuffler logic courtesy of stack overflow.
+//The combination of the two allows the array to be shuffled without original 0th index present.
```

Comment:
Looks good I am curious as to how the second .map plays into the equation here...

Reply:
I'm wondering if there's a way to bring value into the sort method...